packages/video_player_videohole/tizen/src/drm_manager_service_proxy.cc

```diff
@@ -0,0 +1,150 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "drm_manager_service_proxy.h"
+
+#include <dlfcn.h>
+
+#include "log.h"
+
+typedef void (*FuncDMGRSetDRMLocalMode)();
+typedef DRMSessionHandle_t (*FuncDMGRCreateDRMSession)(
+    dm_type_e type, const char* drm_sub_type);
+typedef int (*FuncDMGRSetData)(DRMSessionHandle_t drm_session,
+                               const char* data_type, void* input_data);
+typedef int (*FuncDMGRGetData)(DRMSessionHandle_t drm_session,
+                               const char* data_type, void* output_data);
+typedef bool (*FuncDMGRSecurityInitCompleteCB)(int* drm_handle,
+                                               unsigned int len,
+                                               unsigned char* pssh_data,
+                                               void* user_data);
+typedef int (*FuncDMGRReleaseDRMSession)(DRMSessionHandle_t drm_session);
+typedef int (*FuncPlayerSetDrmHandle)(player_h player,
+                                      player_drm_type_e drm_type,
+                                      int drm_handle);
+typedef int (*FuncPlayerSetDrmInitCompleteCB)(
+    player_h player, security_init_complete_cb callback, void* user_data);
+typedef int (*FuncPlayerSetDrmInitDataCB)(player_h player,
+                                          set_drm_init_data_cb callback,
+                                          void* user_data);
+
+void* Dlsym(void* handle, const char* name) {
+  if (!handle) {
+    LOG_ERROR("[DrmManagerService] dlsym failed, handle is null");
+    return nullptr;
+  }
+  return dlsym(handle, name);
+}
+
+void* OpenDrmManager() { return dlopen("libdrmmanager.so.0", RTLD_LAZY); }
+void* OpenMediaPlayer() {
+  return dlopen("libcapi-media-player.so.0", RTLD_LAZY);
+}
+
+int DMGRSetData(void* handle, DRMSessionHandle_t drm_session,
+                const char* data_type, void* input_data) {
+  FuncDMGRSetData drm_set_data;
+  *(void**)(&drm_set_data) = Dlsym(handle, "DMGRSetData");
+  if (drm_set_data) {
+    return drm_set_data(drm_session, data_type, input_data);
+  }
+  return DM_ERROR_UNKOWN;
+}
```

Comment:
Dynamic loading occurs on every calling DMGRSetData.
I think this is very inefficient. you only need to get the address of the function to dynamically load once and store it somewhere and you can reuse it over and over again during runtime.(Same goes for other similar member functions.)

Reply:
Fixed.